tests/unit/test_config.py

```diff
@@ -141,6 +141,109 @@ def test_maybe_set_compound(monkeypatch, environ, base, name, envvar, expected):
     assert settings == expected
 
 
+@pytest.mark.parametrize(
+    ("environ", "name", "envvar", "coercer", "default", "db", "expected"),
+    [
+        (
+            {"REDIS_URL": "redis://127.0.0.1:6379"},
+            "test.foo",
+            "REDIS_URL",
+            None,
+            None,
+            None,
+            {"test.foo": "redis://127.0.0.1:6379/0"},
+        ),
+        (
+            {"REDIS_URL": "redis://127.0.0.1:6379"},
+            "test.foo",
+            "REDIS_URL",
+            None,
+            None,
+            0,
+            {"test.foo": "redis://127.0.0.1:6379/0"},
+        ),
+        (
+            {"REDIS_URL": "redis://127.0.0.1:6379"},
+            "test.foo",
+            "REDIS_URL",
+            None,
+            None,
+            1,
+            {"test.foo": "redis://127.0.0.1:6379/1"},
+        ),
+        ({}, "test.foo", "REDIS_URL", None, None, None, {}),
+        ({}, "test.foo", "REDIS_URL", None, None, 0, {}),
+        ({}, "test.foo", "REDIS_URL", None, None, 1, {}),
+        (
+            {"REDIS_URL": "redis://127.0.0.1:6379"},
+            "test.foo",
+            "REDIS_URL",
+            str,
+            None,
+            None,
+            {"test.foo": "redis://127.0.0.1:6379/0"},
+        ),
+        (
+            {"REDIS_URL": "redis://127.0.0.1:6379"},
+            "test.foo",
+            "REDIS_URL",
+            str,
+            None,
+            0,
+            {"test.foo": "redis://127.0.0.1:6379/0"},
+        ),
+        (
+            {"REDIS_URL": "redis://127.0.0.1:6379"},
+            "test.foo",
+            "REDIS_URL",
+            str,
+            None,
+            1,
+            {"test.foo": "redis://127.0.0.1:6379/1"},
+        ),
+        ({}, "test.foo", "REDIS_URL", str, None, None, {}),
+        ({}, "test.foo", "REDIS_URL", str, None, 0, {}),
+        (
+            {},
+            "test.foo",
+            "REDIS_URL",
+            str,
+            "redis://127.0.0.1:6379/6",
+            1,
+            {"test.foo": "redis://127.0.0.1:6379/6"},
+        ),
+        (
+            {"REDIS_URL": "redis://127.0.0.1:6379/6"},
+            "test.foo",
+            "REDIS_URL",
+            str,
+            None,
+            9,
+            {"test.foo": "redis://127.0.0.1:6379/9"},
+        ),
+        (
+            {"REDIS_URL": "rediss://foo:bar@example.com:6379/6?fizz=buzz&wu=tang"},
+            "test.foo",
+            "REDIS_URL",
+            str,
+            None,
+            9,
+            {"test.foo": "rediss://foo:bar@example.com:6379/9?fizz=buzz&wu=tang"},
+        ),
+    ],
+)
+def test_maybe_set_redis(
+    monkeypatch, environ, name, envvar, coercer, default, db, expected
```

Comment:
optimization: If `name` can never be `None`, simplify the test cases by one.
`name` is always `"test.foo"`, pass it explicitly in to `config.maybe_set_redis()`
Same with passing the same `envvar` in each test case.

Not a sticking point by any means.

Reply:
aacb95a4a